improver/cli/nowcast_optical_flow_perturbation.py

```diff
@@ -0,0 +1,109 @@
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+"""Script to calculate optical flow advection velocities as perturbations
+from a previous estimate"""
+
+from improver import cli
+
+# Creates the value_converter that clize needs.
+inputadvection = cli.create_constrained_inputcubelist_converter(
+    "precipitation_advection_x_velocity", "precipitation_advection_y_velocity",
+)
+
+
+@cli.clizefy
+@cli.with_output
+def process(
+    current_obs: cli.inputcube,
+    previous_forecast: cli.inputcube,
+    previous_advection: inputadvection,
+    orographic_enhancement: cli.inputcube,
+    *,
+    forecast_period=15,
+):
+    """Calculate optical flow components as a perturbation from the previous
+    advection components, using the difference between a current observation
+    and forecast from the previous time step.
+
+    Args:
+        current_obs (iris.cube.Cube):
+            Cube containing latest radar observation
+        previous_forecast (iris.cube.Cube):
+            Cube containing advection nowcast for this time, created at an
+            earlier timestep
+        previous_advection (iris.cube.CubeList):
+            Advection velocities in the x- and y- directions used to generate
+            the previous forecast
+        orographic_enhancement (iris.cube.Cube):
+            Cube containing orographic enhancement for this timestep
+        forecast_period (int):
+            Forecast period of previous forecast, in minutes.  Used to extract
+            a forecast from a cube that may contain multiple lead times.
+
+    Returns:
+        iris.cube.CubeList:
+            List of the umean and vmean cubes.
+
+    """
+    import iris
+    from iris.cube import CubeList
+
+    from improver.nowcasting.optical_flow import OpticalFlow
+    from improver.nowcasting.utilities import ApplyOrographicEnhancement
+
+    # extract forecast at required lead time (expected in seconds)
+    previous_forecast.coord("forecast_period").convert_units("seconds")
+    previous_forecast = previous_forecast.extract(
+        iris.Constraint(forecast_period=60 * forecast_period)
+    )
+
+    # check that validity times on the forecast and observation match
+    if current_obs.coord("time").cell(0) != previous_forecast.coord("time").cell(0):
+        raise ValueError("Forecast validity time must match input observation")
+
+    # subtract orographic enhancement
+    cube_list = ApplyOrographicEnhancement("subtract")(
+        [current_obs, previous_forecast], orographic_enhancement
+    )
```

Comment:
At the moment, the `cube_list` returned here isn't used when computing the advection perturbations. Should this be, as shown below?
```suggestion
    current_obs, previous_forecast = ApplyOrographicEnhancement("subtract")(
        [current_obs, previous_forecast], orographic_enhancement
    )
```

Reply:
Thanks for spotting this.  I've corrected the error and updated the associated notebook to check that the outputs are "good".